File: php/raindrops/Raindrops.php
<?php declare(strict_types=1);

function raindrops(int $n): string
{
    $r = "";
    if ($n % 3 === 0)
        $r .= "Pling";
    if ($n % 5 === 0)
        $r .= "Plang";
    if ($n % 7 === 0)
        $r .= "Plong";

    return $r ?: (string)$n;
}

File: r/raindrops/raindrops.R
raindrops <- function(number) {
  rain <- list(
    ifelse(number %% 3 == 0, "Pling", ""),
    ifelse(number %% 5 == 0, "Plang", ""),
    ifelse(number %% 7 == 0, "Plong", "")
  )

  rain <- rain[rain != ""]

  if (length(rain) == 0) {
    return(as.character(number))
  }

  paste(rain, collapse="")
}
